distribution/lib/Standard/Table/0.0.0-dev/src/Data/Sort_Column_Selector.enso

```diff
@@ -3,6 +3,6 @@ from Standard.Base import all
 import Standard.Table.Data.Sort_Column
 
 type Sort_Column_Selector
-    type By_Name (columns : Vector Sort_Column.Name) (matcher:Matcher=Text_Matcher)
-    type By_Index (columns : Vector Sort_Column.Index)
-    type By_Column (columns : Vector Sort_Column.Column)
+    type By_Name (columns : Vector (Sort_Column.Name | Text)) (matcher:Matcher=Text_Matcher)
+    type By_Index (columns : Vector (Sort_Column.Index | Integer))
+    type By_Column (columns : Vector (Sort_Column.Column | Column))
```

Comment:
It is a bit worrisome to me that the `Column` type actually does not exist! So we are lying a bit here and it may bite us once we have some validation of these type signatures.

Well - we have a `Column` type but there is one for `Table` and another for `Database` and without interfaces we cannot really relate them. In the absence of typechecking, maybe we can go for `Table.Column`, but in the longer term we probably need to replace this with some `Column` _interface_. This is a common issue across the `Table`/`Database` library.

Reply:
I agree - this is a common issue and one the revised type system will need to resolve. 

Pinging @ekmett - one to consider on the stuff you are working on.